presto-common/src/main/java/com/facebook/presto/common/block/AbstractMapBlock.java

```diff
@@ -432,24 +424,49 @@ void set(int[] hashTables)
         {
             requireNonNull(hashTables, "hashTables is null");
             this.hashTables = hashTables;
+        }
 
-            // The passed in hashTables are always sized as if they are fully built.
-            this.expectedEntryCount = hashTables.length;
+        void setExpectedHashTableCount(int count)
+        {
+            expectedHashTableCount = count;
         }
 
         int getExpectedHashTableCount()
         {
             return expectedHashTableCount;
         }
 
-        public long getInstanceSizeInBytes()
+        public long getRetainedSizeInBytes()
         {
-            return INSTANCE_SIZE;
+            return INSTANCE_SIZE + sizeOf(hashTables);
         }
 
-        public long getRetainedSizeInBytes()
+        public void loadHashTables(int positionCount, int[] offsets, boolean[] mapIsNull, Block keyBlock, MethodHandle keyBlockHashCode)
         {
-            return INSTANCE_SIZE + sizeOfIntArray(expectedEntryCount);
+            int[] hashTables = new int[keyBlock.getPositionCount() * HASH_MULTIPLIER];
+            Arrays.fill(hashTables, -1);
+
+            verify(positionCount < offsets.length, "incorrect offsets size");
+
+            for (int i = 0; i < positionCount; i++) {
+                int keyOffset = offsets[i];
+                int keyCount = offsets[i + 1] - keyOffset;
+                if (keyCount < 0) {
+                    throw new IllegalArgumentException(format("Offset is not monotonically ascending. offsets[%s]=%s, offsets[%s]=%s", i, offsets[i], i + 1, offsets[i + 1]));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;arrayget()&#39; expression, &#39;+&#39; expression, i (declared at line 451). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 455. Consider formatting this data with &#39;%d&#39; instead.